pylint/checkers/classes/class_checker.py

```diff
@@ -887,23 +930,16 @@ def _check_proper_bases(self, node: nodes.ClassDef) -> None:
             ):
                 continue
 
+            if isinstance(ancestor, nodes.ClassDef) and ancestor.is_subtype_of(
+                "enum.Enum"
+            ):
+                self._check_enum_base(node, ancestor)
+
             if not isinstance(ancestor, nodes.ClassDef) or _is_invalid_base_class(
                 ancestor
             ):
                 self.add_message("inherit-non-class", args=base.as_string(), node=node)
 
-            if isinstance(ancestor, nodes.ClassDef) and ancestor.is_subtype_of(
-                "enum.Enum"
-            ):
-                members = ancestor.getattr("__members__")
-                if members and isinstance(members[0], nodes.Dict) and members[0].items:
-                    self.add_message(
-                        "invalid-enum-extension",
-                        args=ancestor.name,
-                        node=node,
-                        confidence=INFERENCE,
-                    )
-
             if ancestor.name == object.__name__:
```

Comment:
It's not directly relevant to this pull request, but I wonder whether moving this `baseclass-is-object` check before the inference steps, with a `continue` statement, might save some time:

```python
        for base in node.bases:
            if getattr(base, 'name', None) == object.__name__:
                self.add_message(
                    "useless-object-inheritance", args=node.name, node=node
                )
                continue
```

Reply:
(in practice: no, this doesn't seem worthwhile to move or adjust at the moment; removing the check entirely doesn't result in a significant difference in the timings produced by `tests/benchmark/test_baseline_benchmarks.py` locally, so I don't think there's a huge opportunity for improvement here)